### PR TITLE
Feat/select

### PR DESCRIPTION
**Pull request by carloshrabelo**
Screen Shot 2020-06-22 at 17 54 03

### CODE REVIEW

File: packages/core/models/select.model.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+export interface ISelect extends HTMLProps<HTMLSelectElement> {
+  /** String */
+  label?: string
+  /** String */
+  error?: string
+  /** String */
+  placeholder?: string
+  /** String */
+  icon?: ReactNode
```
Comment (felipefialho) [inline packages/core/models/select.model.ts:11]: This is specifically to React, use it inside React component ☺️

File: packages/core/models/select.model.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { HTMLProps, ReactNode } from 'react'
```
Comment (felipefialho) [inline packages/core/models/select.model.ts:1]: Inside `core` we need to add some shared things, this is specifically to React, use it inside React component ☺️

File: packages/react-ds/package.json [hunk 3]
```diff
@@ -76,6 +77,7 @@
   "gitHead": "df9fd8986ee04d735ebcd547debf7d6d16ce3b22",
   "dependencies": {
     "@types/classnames": "^2.2.10",
```
Comment (felipefialho) [inline packages/react-ds/package.json:79]: `@types/classnames` is a `devDependencies`

File: packages/styles/components/Select.module.scss [hunk 4]
```diff
@@ -0,0 +1,87 @@
+@import '~@venice/styles/core';
+
+$icon-size: 24px;
+$caret-size: 16px;
+
+.container {
+  @include reset();
```
Comment (felipefialho) [inline packages/styles/components/Select.module.scss:7]: We don't need to use it anymore

File: packages/react-ds/src/components/Select/Select.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import React, { FunctionComponent } from 'react'
+
+import classNames from 'classnames/bind'
+import uniqid from 'uniqid'
+
+import { ISelect } from '@venice/core/models/select.model'
+import styles from '@venice/styles/components/Select.module.scss'
+
+interface ISelectProps extends React.HTMLProps<HTMLSelectElement> {
```
Comment (felipefialho) [inline packages/react-ds/src/components/Select/Select.tsx:9]: Here you can extend a `ISelect`

File: packages/react-ds/src/components/Select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  /** React Element */
+  icon?: React.ReactNode | string
+  /** Style */
+  style?: object
+}
+
+const Select: FunctionComponent<ISelectProps & ISelect> = ({
```
Comment (felipefialho) [inline packages/react-ds/src/components/Select/Select.tsx:16]: And you don't need to use `ISelectProps & ISelect` after extend ISelect later.

File: packages/react-ds/src/components/Select/Select.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  id,
+  label,
+  style,
+  className,
+  error,
+  placeholder,
+  children,
+  icon,
+  ...props
+}: ISelectProps & ISelect) => {
```
Comment (felipefialho) [inline packages/react-ds/src/components/Select/Select.tsx:26]: you don't need to use ISelectProps & ISelect after extend ISelect later.

File: packages/react-ds/src/components/Select/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+      <div className={styles.selectWrapper}>
+        {icon && <span className={styles.icon}>{icon}</span>}
+        <select className={styles.select} id={selfId} {...props}>
+          <option hidden>{placeholder}</option>
+          {children}
+        </select>
+        <div className={styles.caret}>
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 13">
+            <path d="M 23.890625 1.671875 L 23.753906 1.816406 L 13.507812 12.417969 C 13.160156 12.777344 12.640625 13 12.0625 13 C 11.480469 13 10.960938 12.769531 10.613281 12.417969 L 0.375 1.835938 L 0.203125 1.660156 C 0.0742188 1.488281 0 1.285156 0 1.070312 C 0 0.480469 0.554688 0 1.25 0 L 22.855469 0 C 23.550781 0 24.105469 0.480469 24.105469 1.070312 C 24.105469 1.296875 24.023438 1.503906 23.890625 1.671875 Z M 23.890625 1.671875 " />
+          </svg>
```
Comment (felipefialho) [inline packages/react-ds/src/components/Select/Select.tsx:50]: You can add this SVG as a global component (core?) and just import it here (I'll add this caret to SplitButton created by @elizeumatheus)

Comment (carloshrabelo) [reply packages/react-ds/src/components/Select/Select.tsx:50]: I was thinking to get for the icon lib, but i can do a internal workarround

Comment (felipefialho) [reply packages/react-ds/src/components/Select/Select.tsx:50]: Yes, is better to use a icon lib but as we don't have it yet (😢) this icon is a good example to use in both places.

**Pull request merged**